TensorShape.swift file updated to improve TensorShape printing

ksasi opened the pull request:
<!-- What's in this pull request? -->
Added description property to `TensorShape` by conforming to `CustomStringConvertible` so that we can obtain the desired `TensorShape` printing. 

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [TF-447](https://bugs.swift.org/browse/TF-447).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

ksasi committed:
TensorShape.swift file updated to improve TensorShape printing

dan-zheng commented on stdlib/public/TensorFlow/TensorShape.swift:
```diff
@@ -159,3 +159,10 @@ extension TensorShape : Codable {
+  @inlinable
+  public var description: String {
+    return "TensorShape(\(dimensions))"
+    }
```
Please fix indentation:
```suggestion
  }
```

dan-zheng reviewed (approved):
Could you please add a test to `test/TensorFlowRuntime/tensor.swift`?

Something like:
```swift
TensorTests.testAllBackends("TensorShapeDescription") {
  expectEqual("TensorShape([2, 2])", Tensor<Int32>(ones: [2, 2]).shape.description)
  expectEqual("TensorShape([])", Tensor(1).shape.description)
}
```

dan-zheng commented:
A thought: how about using just `[2, 2]` instead of `TensorShape([2, 2])`?

rxwei commented:
I agree that just `[2, 2]` is better.

rxwei commented on stdlib/public/TensorFlow/TensorShape.swift:
```diff
@@ -159,3 +159,10 @@ extension TensorShape : Codable {
 }
+
+extension TensorShape : CustomStringConvertible {
+  @inlinable
```
There's no need for `@inlinable` here.

rxwei commented on stdlib/public/TensorFlow/TensorShape.swift:
```diff
@@ -159,3 +159,10 @@ extension TensorShape : Codable {
+extension TensorShape : CustomStringConvertible {
+  @inlinable
+  public var description: String {
+    return "TensorShape(\(dimensions))"
```
Since we think that `[2, 2]` is more ideal, calling `dimensions.description` should be sufficient.

```suggestion
    return dimensions.description
```

ksasi committed:
TensorShape.swift file updated to improve TensorShape printing
Added test to test/TensorFlowRuntime/tensor.swift

rxwei commented:
@swift-ci please test tensorflow

rxwei commented:
@swift-ci please test tensorflow

ksasi commented:
Updated changes as suggested i.e.
1.Fixed indentation
2.Removed @inlinable
3.Updated "TensorShape(\(dimensions))" to dimensions.description

Also added test to test/TensorFlowRuntime/tensor.swift as below:

TensorTests.testAllBackends("TensorShapeDescription") {
  expectEqual("[2, 2]", Tensor<Int32>(ones: [2, 2]).shape.description)
  expectEqual("[]", Tensor(1).shape.description)
}

rxwei commented:
Thanks!

dan-zheng reviewed (approved):
Thank you!
